package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless

```diff
@@ -207,22 +222,33 @@ util.foreach_radio(uci, function(radio, index, config)
 	uci:set('wireless', radio_name, 'htmode', htmode)
 	uci:set('wireless', radio_name, 'country', site.regdom())
 
+	local outdoor_disabled = nil
+	if (radio.hwmode == '11a' or radio.hwmode == '11na') then
+		if is_outdoor() then
+			uci:set('wireless', radio_name, 'channels', config.outdoor_chanlist())
```

Comment:
According to hostapd.conf if chanlist is unset it will default to all channels allowed in the selected hw_mode:
```
# Channel list restriction. This option allows hostapd to select one of the
# provided channels when a channel should be automatically selected.
# Channel list can be provided as range using hyphen ('-') or individual
# channels can be specified by space (' ') separated values
# Default: all channels allowed in selected hw_mode
#chanlist=100 104 108 112 116
#chanlist=1 6 11-13
```

But I assume that this will break in mac80211.sh here, is that correct?

https://github.com/openwrt/openwrt/blob/master/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh#L97

Reply:
https://github.com/openwrt/openwrt/blob/82625374eea0613027d8dbe2bcfdce7beadaa883/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh#L311 will not add anything if channel_list is empty, so the resulting config will be valid.

Reply:
Okay, but that leaves us with the issue that people with outdoor mode enabled but without a proper chanlist might use an indoor channel outdoors.

Reply:
The only option I see that can't lead to unexpected behaviour when no outdoor channels are defined after a firmware upgrade would to make the outdoor channel list mandatory... or fetch that information from the regdb?

Reply:
I'll take a peek into the latter.

Reply:
Tried iwinfo.nl80211.freqlist, unfortunately it doesn't expose regulatory information. Only `(channel, mhz, restricted)` per channel. No channel was restricted, so I'm assuming this is the live radar state, whether or not that channel has radar activity.

Reply:
The nl80211 interface for regdb information is not exposed through iwinfo and I don't see myself writing that, so making the outdoor channel list mandatory it is.

Reply:
Solved by setting `country3=0x4f` in the hostapd configuration.

```

# The third octet of the Country String (dot11CountryString)
# This parameter is used to set the third octet of the country string.
#
# All environments of the current frequency band and country (default)
#country3=0x20
# Outdoor environment only
#country3=0x4f
# Indoor environment only
#country3=0x49
# Noncountry entity (country_code=XX)
#country3=0x58
# IEEE 802.11 standard Annex E table indication: 0x01 .. 0x1f
# Annex E, Table E-4 (Global operating classes)
#country3=0x04
```